### PR TITLE
handle csi-nodeplugin install and management as a separate step

### PR DESCRIPTION
**Pull request by amarts**
If nodeplugins are restarted upon operator upgrade, there is a
possibility that PVs would be not able to function with FUSE
based filesystems. The process of upgrade (ie, to new version)
would be to update the operator by updating the operator with
newer version, and for client (ie, application/mount point)
upgrades, ideal process is to plan it as combined steps of node
drain, csi-node plugin upgrade, and then mark the node to pick
the applications. This way, there would be no application
dataloss.

Updates: #606
Signed-off-by: Amar Tumballi <amar@kadalu.io>

### CODE REVIEW

**Comment by amarts** [comment]
@Madhu-1 - tagging you here as a reference check on this PR. Have a look when you can spare some time. I referenced your ceph-csi issue for this. Thanks.

**Comment by Madhu-1** [comment]
Thanks, @amarts, sure I will take a look at it soon!

**Comment by amarts** [comment]
note that 'helm' will be broken with this. As they would need to do the step of installing csi-nodeplugin with `kubectl apply`. Will not treat that as a blocker for the issue, as this needs to get in for data sanity during upgrade.

**Comment by leelavg** [comment]
@amarts I'll take care of 'helm' when I'm back (mostly after this PR merge)

I guess we need rolling update for `server` pods as well, what do you feel?

If you don't mind can the PR be opened till Monday EOD?

File: doc/quick-start-yaml.adoc [hunk 1]
```diff
@@ -5,6 +5,7 @@
 [source,console]
 ----
 $ kubectl apply -f https://github.com/kadalu/kadalu/releases/latest/download/kadalu-operator.yaml
+$ kubectl apply -f https://github.com/kadalu/kadalu/releases/latest/download/csi-nodeplugin.yaml
```
Comment (leelavg) [inline doc/quick-start-yaml.adoc:8]: I think, having a note here about why this is decoupled wrt upgrade and node drain would be good.

**Review by leelavg — COMMENTED** [review]
LGTM

File: doc/quick-start-yaml.adoc (hunk 1 above)
Comment (amarts) [reply doc/quick-start-yaml.adoc:8]: #608

**Pull request merged**